source/WeightedList.h

```diff
@@ -137,18 +142,44 @@ typename std::vector<Type>::iterator WeightedList<Type>::eraseAt(typename std::v
 template <class Type>
 typename std::vector<Type>::iterator WeightedList<Type>::erase(typename std::vector<Type>::iterator first, typename std::vector<Type>::iterator last) noexcept
 {
-	auto it = choices.erase(first, last);
+	auto firstWeight = std::next(weights.begin(), std::distance(choices.begin(), first));
+	auto lastWeight = std::next(weights.begin(), std::distance(choices.begin(), last));
+	weights.erase(firstWeight, lastWeight);
 	RecalculateWeight();
-	return it;
+	return choices.erase(first, last);
 }
 
 
 
 template <class Type>
 void WeightedList<Type>::RecalculateWeight()
 {
-	total = std::accumulate(choices.begin(), choices.end(), 0,
-		[](std::size_t x, const Type &t) -> std::size_t { return x + t.Weight(); });
+	total = std::accumulate(weights.begin(), weights.end(), 0, std::plus<int>());
+}
+
+
+
+template <class T, class UnaryPredicate>
+typename std::size_t erase_if(WeightedList<T> &list, UnaryPredicate pred)
```

Comment:
- You don't need the `typename` here.
- This function returns a `std::size_t` while `WeightedList::size` returns an int. Ideally they should both return the same type.

Reply:
> You don't need the `typename` here.

Yup, looks like I erased it up top but not here.